src/mscorlib/shared/System/Globalization/DateTimeParse.cs

```diff
@@ -2295,6 +2295,23 @@ internal static Boolean ProcessHebrewTerminalState(DS dps, ref DateTimeResult re
                         return false;
                     }
                     break;
+                case DS.DX_NNY:
+                    // When formatting, we only format up to the hundred digit of the Hebrew year, although Hebrew year is now over 5000.
+                    // E.g. if the year is 5763, we only format as 763. so we do the reverse when parsing.
```

Comment:
And it's correct that we don't output the thousands place?

Reply:
For Hebrew date formatting, we convert the numbers into words with specific meaning. you may look at 
https://github.com/dotnet/coreclr/blob/master/src/mscorlib/shared/System/Globalization/HebrewNumber.cs#L78

basically, we remove the thousands from the year value before converting it to words/letters. and this is why we return it back when parsing.